odyssey-lib/src/main/java/com/codebutler/odyssey/lib/retro/RetroDroid.kt

```diff
@@ -161,31 +163,33 @@ class RetroDroid(private val context: Context, coreFile: File) : DefaultLifecycl
         if (saveData != null) {
             retro.setMemoryData(Retro.MemoryId.SAVE_RAM, saveData)
         }
-    }
 
-    fun start() {
-        val avInfo = systemAVInfo
-        if (this.timer != null || avInfo == null) {
-            return
-        }
-        this.timer = fixedRateTimer(period = 1000L / avInfo.timing.fps.toLong()) {
-            retro.run()
-        }
+        gameLoaded = true
     }
 
     override fun onResume(owner: LifecycleOwner) {
-        start()
+        Choreographer.getInstance().postFrameCallback(this)
     }
 
     override fun onPause(owner: LifecycleOwner) {
-        stop()
+        Choreographer.getInstance().removeFrameCallback(this)
     }
 
     override fun onDestroy(owner: LifecycleOwner) {
+        Choreographer.getInstance().removeFrameCallback(this)
         unloadGame()
         deinit()
     }
 
+    override fun doFrame(frameTimeNanos: Long) {
```

Comment:
I don't think this is going to work because it assumes that the game framerate is always the same as the device framerate. See for example [PAL SNES games](https://github.com/libretro/snes9x/blob/master/libretro/libretro.cpp#L347). Also, any idea how this API will behave on upcoming 120hz devices?

Reply:
Nevermind this! Misunderstood how this works.

Reply:
Turns out my initial understanding was correct. Calling `retro_run()` twice as fast will run the game at double speed, and this is by design.

I'm going to experiment with an alternate approach that uses a more accurate timer and then only calls `GLSurfaceView#requestRender()` in the Choreographer callback.